src/components/dom.ts

```diff
@@ -670,5 +670,16 @@ export function calculateBaseline(element: Element): number {
  * @param element - The element to toggle the [data-empty] attribute on
  */
 export function toggleEmptyMark(element: HTMLElement): void {
-  element.dataset.empty = Dom.isEmpty(element) ? 'true' : 'false';
+  const isOnlyWhitespace = isCollapsedWhitespaces(element.textContent || '');
+
+  const isElementEmpty = Dom.isEmpty(element) && isOnlyWhitespace;
+
+  element.dataset.empty = isElementEmpty ? 'true' : 'false';
+
+  if (isElementEmpty) {
+    const lastChild = element.lastChild;
+    if (lastChild && lastChild.nodeType === Node.ELEMENT_NODE && (lastChild as HTMLElement).tagName === 'BR') {
+      element.removeChild(lastChild);
+    }
+  }
```

Comment:
I'm not sure that this side effect is good for such a method. Is it really required for fixing the placeholder problem?

Reply:
This cleanup code was added as there was residual `<br>` tag left after deleting data in Header block. Yes, it is better to resolve the problem why there is a `<br>` tag left behind.
I will check the `Header` plugin and update his PR.

Reply:
I thought that problem with placeholder is related to whitespace symbols, but not "br". Our "isEmpty" treats elements containing only "br" as empty, so there should not be a problem